lib/src/htmlfilesetserv/HTMLFileSetHTTPServer.java

```diff
@@ -687,19 +641,22 @@ private AuthToken getToken(final HttpServletRequest request)
 		}
 		if (request.getCookies() != null) {
 			for (final Cookie c: request.getCookies()) {
-				if (c.getName().equals(TOKEN_COOKIE_NAME) &&
-						!c.getValue().isEmpty()) {
-					return auth.validateToken(
-							unmungeCookiePerShane(c.getValue()));
+				if (c.getName().equals(TOKEN_COOKIE_NAME) && !c.getValue().isEmpty()) {
+					return auth.validateToken(unmungeCookiePerShane(c.getValue()));
 				}
 			}
 		}
+		// hacky hack hack - remove later when a final solution for cookies is found
+		for (final Cookie c: request.getCookies()) {
```

Comment:
This second TOKEN_COOKIE2_NAME case should go inside `if (request.getCookies() != null) {}` block (one code line up).